search/utils.py

```diff
@@ -0,0 +1,366 @@
+    Args:
+        resource (LearningResource): Item to convert to dict.
+        term_info (dict): Vocabulary terms assigned to resource.
+    Returns:
+        rec (dict): Dictionary representation of the LearningResource.
+    """
+
+    if term_info is None:
+        term_info = {}
+    rec = {
+        "title": resource.title,
+        # The zero is for sorting blank items to the bottom. See below.
+        "titlesort": "0{0}".format(resource.title.strip()),
+        "id": resource.id,
+        "_id": resource.id,  # The ID used by Elasticsearch.
+        "resource_type": resource.learning_resource_type.name,
+        "description": resource.description,
+        "content_xml": resource.content_xml,
+        "content_stripped": strip_xml(resource.content_xml),
+        "xa_nr_views": resource.xa_nr_views,
+        "xa_nr_attempts": resource.xa_nr_attempts,
+        "xa_avg_grade": resource.xa_avg_grade,
+        "xa_histogram_grade": resource.xa_histogram_grade,
+    }
+
+    course = get_course_metadata(resource.course_id)
+    rec["preview_url"] = get_preview_url(
+        resource,
+        org=course["org"],
+        course_number=course["course_number"],
+        run=course["run"],
+    )
+    rec["run"] = course["run"]
+    rec["course"] = course["course_number"]
+    rec["repository"] = course["repo_slug"]
+
+    # Index term info. Since these fields all use the "not_analyzed"
+    # index, they must all be exact matches.
+    for key, vals in term_info.items():
+        rec[key] = vals
+
+    # If the title is empty, sort it to the bottom. See above.
+    if rec["titlesort"] == "0":
+        rec["titlesort"] = "1"
+
+    # Keys that may have unicode issues.
+    text_keys = (
+        'title', 'titlesort', 'resource_type', 'description', 'content_xml',
+        'content_stripped',
+    )
+    for key in text_keys:
+        try:
+            rec[key] = rec[key].decode('utf-8')
+        except AttributeError:
+            pass  # Python 3
+
+    return rec
+
+
+def clear_index():
+    """Wipe the index."""
+    if CONN.indices.exists(INDEX_NAME):
+        CONN.indices.delete(INDEX_NAME)
+        CONN.indices.create(INDEX_NAME)
+        CONN.indices.refresh()
+    create_mapping()
+
+    # re-index all existing LearningResource instances:
+    index_resources(LearningResource.objects.all())
+
+
+class SearchResults(object):
+    """
+    Helper class for elasticsearch_dsl search results.
+
+    This is because the actual search result has nested types, which
+    would be cumbersome to access from arbitrary calling functions.
+    """
+    def __init__(self, search):
+        """Get raw search result from Elasticsearch."""
+        self._search = search
+
+    def count(self):
+        """Total records matching the query."""
+        return self._search.count()
+
+    def page_count(self):
+        """Total number of result pages."""
+        count = self._search.count() / PAGE_LENGTH
+        if self._search.count() % PAGE_LENGTH > 0:
```

Comment:
In this case we should use a local variable for `self._search.count()` so that the page count returned is consistent with a single count value. If `self._search.count()` returns two different numbers then the page count calculation doesn't make sense.